### PR TITLE
[iDEAL] Bank Selection Logic

### PR DESCRIPTION
**Pull request by mpetrenco**
- Connected the iDEAL Form with the Bank Selection page;
- Added 'Pay' button enable/disable logic based on the selected bank;
- Added 'iDEAL Transaction' option on the example app;

### CODE REVIEW

File: Source/iDEAL/IDEALBankSelectionTableViewController.m [hunk 1]
```diff
@@ -78,4 +78,12 @@ - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPa
     return 60.0;
 }
 
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
+    IDEALBank *bank = self.bankList[indexPath.row];
+    if (self.delegate) {
```
Comment (stpopa) [inline Source/iDEAL/IDEALBankSelectionTableViewController.m]: `[self.delegate responsedToSelector: @selector(didSelectBank:)]`

Comment (mpetrenco) [reply Source/iDEAL/IDEALBankSelectionTableViewController.m]: `self.delegate` is of type `id<IDEALBankSelectionDelegate>`.
I won't be able to call `respondsToSelector:` without casting the delegate to an NSObject or such.

Comment (stpopa) [reply Source/iDEAL/IDEALBankSelectionTableViewController.m]: just `@protocol IDEALBankSelectionDelegate <NSObject>`

Comment (mpetrenco) [reply Source/iDEAL/IDEALBankSelectionTableViewController.m]: Done.

File: Source/iDEAL/IDEALBankSelectionTableViewController.h [hunk 2]
```diff
@@ -24,10 +24,32 @@
...
+ * A delegate protocol related to the bank selection logic
+ */
+@protocol IDEALBankSelectionDelegate <NSObject>
+
+/**
+ * A delegate method that is called when a bank from the iDEAL bank list is selected
+ *
+ * @param bank - the iDEAL bank model containing the bank title and identifier code
+ */
+- (void)didSelectBank:(nonnull IDEALBank *)bank;
```
Comment (stpopa) [inline Source/iDEAL/IDEALBankSelectionTableViewController.h]: - (void)viewController:(IDEALBankSelectionTableViewController *) controller didSelectBank:(nonnull IDEALBank *)bank;

Comment (stpopa) [reply Source/iDEAL/IDEALBankSelectionTableViewController.h]: `To make it clear that a method is indeed a delegate method, it's common practice to start the method name with the name of the type that is delegating - like how every method above starts with fileImporter.`

https://www.swiftbysundell.com/articles/delegation-in-swift/

Comment (mpetrenco) [reply Source/iDEAL/IDEALBankSelectionTableViewController.h]: Will be done in the latest PR

**Pull request merged**